packages/jest-runtime/src/index.ts

```diff
@@ -1260,14 +1265,26 @@ export default class Runtime {
       throw new Error('You need to call `stopCollectingV8Coverage` first.');
     }
 
+    const coverageMatcher = this._coverageOptions.collectCoverageFrom.length
+      ? globsToMatcher(this._coverageOptions.collectCoverageFrom)
+      : undefined;
+    const matchesCoveragePattern = (file: string) => {
+      if (coverageMatcher) {
+        return coverageMatcher(
+          replacePathSepForGlob(path.relative(this._config.rootDir, file)),
+        );
+      }
+      return this._v8CoverageSources!.has(file);
+    };
+
     return this._v8CoverageResult
       .filter(res => res.url.startsWith('file://'))
       .map(res => ({...res, url: fileURLToPath(res.url)}))
       .filter(
         res =>
           // TODO: will this work on windows? It might be better if `shouldInstrument` deals with it anyways
           res.url.startsWith(this._config.rootDir) &&
-          this._v8CoverageSources!.has(res.url) &&
+          matchesCoveragePattern(res.url) &&
           shouldInstrument(res.url, this._coverageOptions, this._config),
```

Comment:
hmmm, thinking about it - shouldn't this option be enough? and the fact the css transform is included is because we are missing an ignore pattern in our test?

(sorry about the back and forth 🙈)

Reply:
No worries at all. I'm really not familiar with the code base or v8 so I don't fully understand what's going on.
I was simply following the instructions from someone who answered my [question](https://stackoverflow.com/questions/75342970/is-it-possible-to-collect-coverage-on-code-loaded-with-vm-runincontext-with-jest) on StackOverflow.

I `blamed` the file and I see that `_v8CoverageSources` was introduced as a substitute to `_fileTransforms ` in https://github.com/facebook/jest/pull/12912 to avoid losing the list of files when calling `jest.resetModules()`.

However, I don't see why we would need this in the filter. The 2 other checks should be enough (`es.url.startsWith(this._config.rootDir) && shouldInstrument(...)`. This is why this PR was removing it: https://github.com/facebook/jest/pull/10657

So if you agree, I will remove this line and exclude `cssTransform.js` in the other test so that it doesn't appear in the coverage. Could you confirm this is the correct way?

Reply:
For me it looks right if by default only files explicitly imported in tests are included in the coverage report. For example, that’s the default of [`c8`](https://github.com/bcoe/c8). They have `all`, `include`, `exclude` option to include other files.

So it made sense that `collectCoverageFrom` option is collecting coverage from not only imported files. Current Jest documentation of the option agrees: "If a file matches the specified glob pattern, coverage information will be collected for it even if no tests exist for this file and it's never required in the test suite."

If I didn't miss some detail.

Reply:
So maybe if I can't find the file in _v8CoverageSource, I can call `transformFile()` on it and add it to the map. Would that be ok?

Reply:
Not sure what you have in mind. I was trying to say that not including `cssTransform.js` in coverage report by default seems to be expected behaviour.

Reply:
@mrazauskas so do you mean we shouldn't touch that part of the code and find another way to add the files loaded with `vm.runInContext` to `_v8CoverageSources`?

Reply:
Let’s wait for Simen’s feedback. He was commenting that `cssTransform.js` could get included or not because of ignore pattern. I try to suggest that by default only explicitly imported files should be included in the report and `collectCoverageFrom` option could be used to override that. This is what your current implementation does. If I got it right.

Reply:
My thought was that `shouldInstrument` should return the right thing - if a file shouldn't get instrumented, it shouldn't be include in a coverage report

Reply:
@SimenB you're right, my change didn't make sense.
I've removed the line `this._v8CoverageSources!.has(res.url)` and added `cssTransform.js` to the `coveragePathIgnorePatterns` in the other tests. I hope that's ok.

Reply:
Right, this seems more reasonable to me 👍